### PR TITLE
DOP-4511: add link with community-build badge component

### PR DESCRIPTION
**Pull request by schmalliso**
### Stories/Links:

Jira: DOP-4511

### Current Behavior:

This is net new!

### Staging Links:

*new* https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master2/landing/allison/DOP-4511/languages/example/index.html

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Comment by schmalliso** [comment]
There's something slightly odd when you first load the page, where the badge seems to shift from a newline back up and I have no idea why, so that's fun.

File: src/components/Roles/CommunityPillLink.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import PropTypes from 'prop-types';
+import Badge from '@leafygreen-ui/badge';
+import Link from '../Link';
+
+const CommunityPillLink = ({
+  nodeData: {
+    children: [{ value }],
+  },
+}) => {
+  let [label, link] = value.split('<');
```
Comment (seungpark) [inline src/components/Roles/CommunityPillLink.js:11]: question for self. from [this](https://github.com/mongodb/snooty-parser/pull/585/files) how does one know to split by < ?

**Review by seungpark — COMMENTED** [review]
this LGTM 👍 

would add a small unit test to store the snapshot before merge 👍

**Comment by rayangler** [comment]
> There's something slightly odd when you first load the page, where the badge seems to shift from a newline back up and I have no idea why, so that's fun. Possibly related to [DOP-4501](https://jira.mongodb.org/browse/DOP-4501)?

Yeah, it's super weird. Seems like the Badge component is initially rendered differently in the HTML (not within the span). This results in hydration errors in the console, which I don't think is ideal for SEO. I honestly can't tell why it's happening though.

Screen Shot 2024-04-16 at 12 16 10 PM

File: src/components/Roles/CommunityPillLink.js (hunk 1 above)
Comment (schmalliso) [reply src/components/Roles/CommunityPillLink.js:11]: Correct, that's consistent with every other link-y role.

File: src/components/Roles/CommunityPillLink.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  if (link) {
+    link = link.split('>')[0];
+    label = label.trim();
+  }
+
+  return (
+    <span>
```
[the remark this answers is not in the document]
Comment (schmalliso) [reply src/components/Roles/CommunityPillLink.js:18]: I'm concerned folks will want to use it inline, but perhaps we can just… not let them?

[the remark this answers is not in the document]
Comment (schmalliso) [reply src/components/Roles/CommunityPillLink.js:18]: I'll convert to a directive and refactor accordingly to potentially avoid the layout shift issue. Will let Caitlin et al. know they can't use it inline.

**Comment by schmalliso** [comment]
Updated to use a directive rather than role, which seems to have addressed the weird loading behavior / sad console errors. (I do still need to add the wee test.)

**Review by seungpark — APPROVED** [review]
nice switch to directive! was going to mention the block element but seems like its already being addressed 👍

**Pull request merged**